twilly_cli/src/sync/maps.rs

```diff
@@ -80,6 +89,187 @@ pub async fn choose_map_action(twilio: &Client, sync_service: &SyncService) {
                     println!("{:#?}", selected_sync_map);
                     println!();
                 }
+                Action::Rename => {
+                    let get_name_prompt = Text::new(
+                        "What would you like to rename this map to? Must be supported characters '^[a-zA-Z0-9-_]+$'"
+                    );
+                    let get_name_result = prompt_user(get_name_prompt);
+
+                    if let None = get_name_result {
+                        break;
+                    }
+
+                    let allowed_chars = Regex::new(r"^[a-zA-Z0-9-_]+$").unwrap();
+                    let name = get_name_result.unwrap();
+                    let trimmed_name = name.trim();
+                    if !allowed_chars.is_match(trimmed_name) {
+                        println!("Name doesn't match required filter '^[a-zA-Z0-9-_]+$'");
+                        break;
+                    }
+
+                    println!("Name confirmed '{trimmed_name}'");
+
+                    let confirmation_message = "⚠️ Warning ⚠️
+
+This process is non-reversible. We will:
+    1. Create a temporary map to hold a copy of the map items
+    2. Copy the items into the temporary map
+    3. Confirm the copy worked
+    4. Delete the original map
+    5. Create a new map with your new name
+    6. Copy all items from the temporary map into the new map
+
+We will not delete the temporary map after the process has completed.
+You can remove this using the CLI after you've confirmed the rename was successful.
+Would you like to continue?";
+                    let confirm_operation = Confirm::new(confirmation_message)
+                        .with_placeholder("N")
+                        .with_default(false);
+
+                    let confirmation_result = prompt_user(confirm_operation);
+
+                    if let None = confirmation_result {
```

Comment:
Checking for `None` here is correct to catch a cancelled action and return the user to the previous stage however this will not catch the user selecting _no_ and hence currently any input will pass through. [See explicit check here](https://github.com/TristanBlackwell/twilly/blob/2d24dd746771b5cb563d23b65ca4e5213572acd4/twilly_cli/src/conversation.rs#L540C4-L541C34).

Reply:
Thanks, resolved 👍